source/index.md.erb

```diff
@@ -41,1706 +114,2044 @@ curl https://nuv.la?media=xml
+  a new SSH key pair.
 
-```xml
-<runs offset="0" limit="20" count="20" totalCount="93">
-   <item resourceUri="run/fed1dd04-aa05-4a92-abb8-ed279f1f4dd4" uuid="fed1dd04-aa05-4a92-abb8-ed279f1f4dd4" moduleResourceUri="module/SlipStream/deploy-slipstream-dpl/745" status="Done" abort="" startTime="2015-03-17 13:34:51.440 UTC" cloudServiceNames="exoscale-ch-gva" username="sixsq_dev" type="Orchestration" tags=""/>
-   <item resourceUri="run/1ab29b6a-f08c-45ae-9cf4-336a0415f46c" uuid="1ab29b6a-f08c-45ae-9cf4-336a0415f46c" moduleResourceUri="module/SlipStream/deploy-slipstream-dpl/742" status="Cancelled" abort="Failed running &apos;execute&apos; target on &apos;slipstream.1&apos;" startTime="2015-03-17 10:52:05.331 UTC" cloudServiceNames="exoscale-ch-gva" username="sixsq_dev" type="Orchestration" tags=""/>
-   ...
-</runs>
+```json
+{
+  "credentialTemplate" : {
+                           "href" : "credential-template/generate-ssh-key-pair"
+                         }
+}
 ```
 
-List all runs of a user.
-
-### HTTP Request
-
-`GET https://nuv.la/run`
-
-## Get a Run
-
 ```shell
-curl https://nuv.la/run/<run> -u <user>:<password>
-```
-
-```xml
-<runs>
-   <item resourceUri="run/4f187379-cdb9-470d-9656-b6fac6dc33d8" uuid="4f187379-cdb9-470d-9656-b6fac6dc33d8" moduleResourceUri="module/examples/tutorials/service-testing/system/471" status="Done" startTime="2014-05-10 15:22:56.882 UTC" cloudServiceName="exoscale-ch-gva" username="meb" type="Orchestration" tags=""/>
-   <item resourceUri="run/fa3b9652-9dac-49e2-9573-b761777c8238" uuid="fa3b9652-9dac-49e2-9573-b761777c8238" moduleResourceUri="module/examples/tutorials/service-testing/system/471" status="Done" startTime="2014-05-10 15:13:29.352 UTC" cloudServiceName="exoscale-ch-gva" username="meb" type="Orchestration" tags=""/>
-</runs>
-```
-
-Get a specific run.
-
-### HTTP Request
+# Be sure to get the URL from the cloud entry point!
+# The cookie options allow for automatic management of the
+# SlipStream authentication token (cookie).
+curl https://nuv.la/api/credential \
+     -X POST \
+     -H 'content-type: application/json' \
+     -d @create.json \
+     --cookie-jar ~/cookies -b ~/cookies -sS
+```
+
+> On a successful authentication, the above command will return a 201
+  (created) status, a "location" header with the created credential
+  resource, and a JSON document containing the SSH private key of the
+  generated key pair.
+
+The "ssh-public-key" Credential resource stores the public key (in
+OpenSSH format), the algorithm used to create the key ("rsa" or
+"dsa"), and the fingerprint of the key itself.
+
+These resources can be created either by providing the public key of
+an existing SSH key pair or by having the server generate a new SSH
+key pair.  In the second case, the server will provide the private key
+in the 201 (created) response.
 
-`GET https://nuv.la/run/<run-uuid>`
+<aside class="notice">
+When the server generates a new SSH key pair, the server returns the
```

Comment:
You can create any number of SSH public key credentials on the server.  I've added a sentence about this.